L/libjulia/common.jl

```diff
@@ -0,0 +1,161 @@
+    USE_SYSTEM_LIBUNWIND=1
+    USE_SYSTEM_PCRE=1
+    USE_SYSTEM_OPENLIBM=1
+    USE_SYSTEM_DSFMT=1
+    USE_SYSTEM_BLAS=1
+    LIBBLASNAME=libopenblas
+    USE_SYSTEM_LAPACK=1
+    LIBLAPACKNAME=libopenblas
+    USE_SYSTEM_GMP=1
+    USE_SYSTEM_MPFR=1
+    USE_SYSTEM_SUITESPARSE=1
+    USE_SYSTEM_LIBUV=1
+    USE_SYSTEM_UTF8PROC=1
+    USE_SYSTEM_MBEDTLS=1
+    USE_SYSTEM_LIBSSH2=1
+    USE_SYSTEM_CURL=1
+    USE_SYSTEM_LIBGIT2=1
+    USE_SYSTEM_PATCHELF=1
+    USE_SYSTEM_ZLIB=1
+    USE_SYSTEM_P7ZIP=1
+
+    override XC_HOST=${target}
+    override OS=${OS}
+
+    #llvm-config-host is not available
+    override LLVMLINK=-L${prefix}/lib -lLLVM-9jl
+    override LLVM_CXXFLAGS=-I${prefix}/include -std=c++14 -fno-exceptions -fno-rtti -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS
+    override LLVM_LDFLAGS=-L${prefix}/lib
+
+    # just nop this
+    override LLVM_CONFIG_HOST=
+
+    # julia expects libuv-julia.a
+    override LIBUV=${prefix}/lib/libuv.a
+
+    prefix=${prefix}
+    LOCALBASE=${prefix}
+    EOM
+
+    make BUILDING_HOST_TOOLS=1 NO_GIT=1 -j${nproc} -C src/flisp host/flisp
+    make clean -C src
+    make clean -C src/support
+    make clean -C src/flisp
+    make USE_CROSS_FLISP=1 NO_GIT=1 -j${nproc} julia-ui-release
+
+    # compile libjulia but don't try to build a sysimage
+    mkdir -p ${libdir}
+    mkdir -p ${includedir}/julia
+    cp usr/lib/libjulia* ${libdir}/
+    cp -R -L usr/include/julia/* ${includedir}/julia
+    install_license LICENSE.md
+    """
+
+    # These are the platforms we will build for by default, unless further
+    # platforms are passed in on the command line
+    platforms = expand_cxxstring_abis(supported_platforms())
+
+    # The products that we will ensure are always built
+    products = [
+        LibraryProduct("libjulia", :libjulia; dont_dlopen=true),
+    ]
+
+    # Dependencies that must be installed before this package can be built/used
+    dependencies = [
+        Dependency("LibUnwind_jll"),
+        Dependency("LibOSXUnwind_jll"),
+        Dependency(PackageSpec(name="PCRE2_jll", version=v"10.31")),
+        Dependency("OpenLibm_jll"),
+        Dependency("dSFMT_jll"),
+        Dependency(PackageSpec(name="SuiteSparse_jll", version=v"5.4.0")),
+        Dependency("LibUV_jll"),
+        Dependency("utf8proc_jll"),
+        Dependency("MbedTLS_jll"),
+        Dependency("LibSSH2_jll"),
+        Dependency("LibCURL_jll"),
+        Dependency("Patchelf_jll"),
+        Dependency("Zlib_jll"),
+        Dependency("p7zip_jll"),
+    ]
+    if version.major == 1 && version.minor == 4
+        push!(dependencies, Dependency(PackageSpec(name="OpenBLAS_jll", version=v"0.3.5")))
+        push!(dependencies, Dependency(PackageSpec(name="libLLVM_jll", version=v"8.0.1")))
+        push!(dependencies, Dependency(PackageSpec(name="MPFR_jll", version=v"4.0.2")))
+        push!(dependencies, Dependency(PackageSpec(name="GMP_jll", version=v"6.1.2")))
```

Comment:
As discussed elsewhere, unfortunately this doesn't work on Yggdrasil, as it uses Julia 1.6dev and that means we can't load GMP_jll.

But I also don't think we actually need to install it here? We can just let Julia download the correct BB artifacts? That seemed to work well enough in PR #1664?

Reply:
A fix is in the works https://github.com/JuliaLang/Pkg.jl/pull/2031, and until then I plan to just do a local deploy.

The benefit for using the proper way of downloading the dependencies is that you get caching.

Reply:
Yes, but the drawback is that it doesn't work in some cases, e.g. for GMP_jll. So at least for those JLLs which limit the compatible Julia versions, I recommend not to try to use the JLL as above (until BB / Pkg.jl learn how to deal with this)

Reply:
Right, I am happy to deploy from my local build and not Yggdrasil for now, until Pkg learns to handle this. 
But I am I prefer to fix "easy" issues like that instead of working around them. If we really want to deploy from Yggdrasil, @staticfloat might be able to deploy an experimental version with that patch included.

Reply:
Well, in the case of libjulia, though, the fix is pretty easy, so why not go for it? Namely, just don't use GMP_jll, and instead ask Julia to download it. That works pretty well in PR #1664.